README.md

```diff
@@ -17,13 +37,112 @@ var completed = sourceList.filter(function(todo){
 });
 ```
 
-Any changes to `sourceList` will automatically update the derived `completed` list in O(log n) time.  
-This avoids the standard O(n) times done everywhere else.
+Any changes to `sourceList` will automatically update the derived `completed` list:
+```
+completed.bind('add', function(ev, newItems){
+	console.log(newItems);
+});
+
+sourceList.push({name: "cook", complete: true}, {name: "clean", complete: false});
+// `{name: "cook", complete: true}`
+```
+
+### With can.Map.define
+
+If you're using the [can.Map.define plugin](http://canjs.com/docs/can.Map.prototype.define.html), you can define a derived list like so:
+
+```
+{
+	define: {
+		todos: {
+			Value: can.List
+		},
+		completedTodos: {
+			value: function(){
+				return this.attr('todos').dFilter(function(todo){
+					return todo.attr('complete');
+				});
+			}
+		}
+	}
+}
+```
+
+Note that you are using `value` and not `get`. With `get`, every time `todos` changes, a brand new derived list will be generated. Since the derived list updates itself based on the source list, make sure only to instantiate the derived list once.
+
+## API
+
+The `can-derive` plugin adds the `dFilter` method to `can.List.prototype`, so all filtered lists are derived from a `can.List`.
+
+### dFilter
+
+`list.filter(predicate) -> DerivedList`
+
+Generates a derived list based on a predicate function.
+
+### attr
+
+`derivedList.attr() --> Array`
+
+Gets an array of all the elements in the derived list.
+
+`derivedList.attr(index) --> Object`
+
+Reads an element from an index on the filtered list. This references the *element in the source list* that *is at the specified index in the filtered list*.
+
+### each
+
+`derivedList.each(fn) --> DerivedList`
+
+Iterates through the DerivedList, calling a function for each element. The elements provided inside the function will be the elements *in the source list*. The filtered list will be updated only if the outcome of the predicate function changes.
+
+### Other can.List methods
+
+Since DerivedList is a [can.List](http://canjs.com/docs/can.List.html), the following methods are also available:
+
+- [filter](http://canjs.com/docs/can.List.prototype.filter.html)
+- [indexOf](http://canjs.com/docs/can.List.prototype.indexOf.html)
+- [map](http://canjs.com/docs/can.List.prototype.map.html)
+- [replace](http://canjs.com/docs/can.List.prototype.replace.html)
+- [slice](http://canjs.com/docs/can.List.prototype.slice.html)
+
+### Disabled can.List methods
+
+The filtered list is not changed manually, but is maintained as the source list changes. Because of this, the following `can.List` methods are disabled:
+
+- push
+- pop
+- shift
+- unshift
+- splice
+
+## Performance
+
+`can-derive` optimizes for indertions and removals, completing them in O(log n) time. This means that changes to the source list will automatically update the derived list in O(log n) time, compared to the standard O(n) time you would expect in other implementations.
```

Comment:
Typo: "indertions" > "insertions"

Also, I think the "O(log n)" bits should be code formatted like, `O(log n)`. But I may be wrong about that.